staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -45,7 +45,7 @@ import (
 var directDialer utilnet.DialFunc = http.DefaultTransport.(*http.Transport).DialContext
 
 func init() {
-	client.Metrics.RegisterMetrics(compbasemetrics.NewKubeRegistry().Registerer())
+	client.Metrics.RegisterMetrics(legacyregistry.Registerer())
```

Comment:
is there any test that can demonstrate the old way did not work and this new way does?

Reply:
We should have unit tests for this but I'm okay with doing that in a separate PR so that this can more easily be backported.

Reply:
asked another way, how do we know this is doing what it should now and should be backported?

Reply:
@liggitt 

I tried a couple unit test approaches that fell short:
* need to mock unix socket, a gRPC service, and still have to live with some hardcoded 30s timeouts...
* interrogate the legacy registry directly (GatherAndLint?). I couldn't find a test helper that would detect regression.

@logicalhan pointed me to http://go/kubernetes-main/staging/src/k8s.io/component-base/metrics/prometheus/feature/metrics.go#L47-L53. This approach seems much nicer but will need a new konnectivity-client feature metric to be used here. If you agree I like this approach and can do it separately.

Reply:
I'm not suggesting creating a new metric, I'm suggesting exposing publicly the method for recording data into a metric. That way, you can just unit test that function and verify that metric data is written out to the endpoint.

Reply:
> Oh, @jkh52 tested the endpoint manually.

That's right, although I agree with @liggitt about wanting regression test coverage of this bugfix.

> I'm suggesting exposing publicly the method for recording data

Those methods are in https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/konnectivity-client/pkg/client/metrics/metrics.go. The instrumented metrics have coverage at https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/konnectivity-client/pkg/client/client_test.go

I think all we're looking for here is that client metrics are registered with the legacy registry.

Reply:
> Why can't you just import ClientMetrics and invoke ObservePacket and then verify it is in the legacyregistry endpoint?

Yeah, this was sort of what I was expecting

Reply:
ah - this seems feasible

/hold